packages/devtools_app/lib/main.dart

```diff
@@ -6,9 +6,49 @@ import 'package:flutter/material.dart';
 
 import 'src/app.dart';
 import 'src/config_specific/framework_initialize/framework_initialize.dart';
+import 'src/config_specific/load_fallback_app/load_fallback_app.dart';
 import 'src/preferences.dart';
 
 void main() async {
+  final defaultOnError = FlutterError.onError;
+  var numErrors = 0;
+  var fallbackVersionDialogShown = false;
+
+  // The SKIA/CanvasKit version of Flutter Web is newer than other platforms
+  // and depends on WebGL so we need to detect whether it is behaving badly
+  // to give users the option to reopen DevTools using a fallback slower version
+  // that may be more stable.
+  // TODO(jacobr): remove this error listening code once
+  // https://github.com/flutter/devtools/issues/2125 is fixed.
+  if (const bool.fromEnvironment('FLUTTER_WEB_USE_SKIA')) {
+    FlutterError.onError = (FlutterErrorDetails details) {
+      if (defaultOnError != null) {
+        defaultOnError(details);
+      }
+      final bool overflowError = details.toString().contains('overflowed');
+
+      if (overflowError) {
+        // Overflow errors do not indicate a CanvasKit specific problem and
+        // occur frequently in debug builds of apps so we filter them out to
+        // avoid distraction. We could alternately suppress this code completely
+        // in debug builds but it is risky to add code that runs in release
+        // builds but not debug builds.
+        return;
+      }
+      // TODO(jacobr): detect SKIA specific errors and warn on the first SKIA
```

Comment:
@hterkelsen, remind me, is there a publicly visible type I can check for a Skia specific error?